catppuccin.tmux

```diff
@@ -31,6 +31,160 @@ setw() {
   tmux_commands+=(set-window-option -gq "$option" "$value" ";")
 }
 
+build_window_icon() {
+  local window_status_icon_enable="$(get_tmux_option "@catppuccin_window_status_icon_enable" "yes")"
+
+  local custom_icon_window_last="$(get_tmux_option "@catppuccin_icon_window_last" "󰖰")"
+  local custom_icon_window_current="$(get_tmux_option "@catppuccin_icon_window_current" "󰖯")"
+  local custom_icon_window_zoom="$(get_tmux_option "@catppuccin_icon_window_zoom" "󰁌")"
+  local custom_icon_window_mark="$(get_tmux_option "@catppuccin_icon_window_mark" "󰃀")"
+  local custom_icon_window_silent="$(get_tmux_option "@catppuccin_icon_window_silent" "󰂛")"
+  local custom_icon_window_activity="$(get_tmux_option "@catppuccin_icon_window_activity" "󰖲")"
+  local custom_icon_window_bell="$(get_tmux_option "@catppuccin_icon_window_bell" "󰂞")"
+
+  if [[ $window_status_icon_enable == "yes" ]]
+  then
+    local show_window_status="#(printf '%%s\n' '#F' | sed \"s/*/${custom_icon_window_current}/\" | sed \"s/-/${custom_icon_window_last}/\" | sed \"s/#/${custom_icon_window_activity}/\" | sed \"s/#//g\"| sed \"s/~/${custom_icon_window_silent}/\" | sed \"s/!/${custom_icon_window_bell}/\" | sed \"s/M/${custom_icon_window_mark}/\" | sed \"s/Z/${custom_icon_window_zoom}/\")"
```

Comment:
This can be done with a single sed invocation like this: `sed 's/../../; s/../../'`

Also note that sed will only substitute the first match in a line by default, you have to write `s/../../g` to substitute all matches in a line. I’m not sure which one you want here, just letting you know.

Reply:
I am assuming here the the characters i am substituting will only appear once in the line.